rdtools/availability.py

```diff
@@ -0,0 +1,620 @@
+    and one for system-wide outages. The :py:meth:`.AvailabilityAnalysis.run()`
+    method executes both algorithms and combines their results.
+
+    The input timeseries don't need to be in any particular set of units as
+    long as all power and energy units are consistent, with energy units
+    being the hourly-integrated power (e.g., kW and kWh). The units of the
+    analysis outputs will match the inputs.
+
+    Parameters
+    ----------
+    power_system : pd.Series
+        Timeseries total system power. In the typical case, this is meter
+        power data.
+
+    power_subsystem : pd.DataFrame
+        Timeseries power data, one column per subsystem. In the typical case,
+        this is inverter AC power data. Each column is assumed to represent
+        a subsystem, so no extra columns may be included. The index must
+        match ``power_system``.
+
+    energy_cumulative : pd.Series
+        Timeseries cumulative energy data for the entire system (e.g. meter).
+        These values must be recorded at the device itself (rather than summed
+        by a downstream device like a datalogger or DAS provider) to preserve
+        its integrity across communication interruptions. Units must match
+        ``power`` integrated to hourly energy (e.g. if ``power`` is in kW then
+        ``energy`` must be in kWh).
+
+    power_expected : pd.Series
+        Expected system power data with the same index as the measured data.
+        This can be modeled from on-site weather measurements if instruments
+        are well calibrated and there is no risk of data gaps. However, because
+        full system outages often cause weather data to be lost as well, it may
+        be more useful to use data from an independent weather station or
+        satellite-based weather provider.
+
+    Attributes
+    ----------
+    results : pd.DataFrame
+        Rolled-up production, loss, and availability metrics. The index is
+        a datetime index of the period passed to
+        :py:meth:`AvailabilityAnalysis.run`. The columns of the dataframe are
+        as follows:
+
+        +----------------------+----------------------------------------------+
+        | Column Name          | Description                                  |
+        +======================+==============================================+
+        | 'lost_production'    | Production loss from outages. Units match the|
+        |                      | input power units (e.g. if power is given in |
+        |                      | kW, 'lost_production' will be in kWh).       |
+        +----------------------+----------------------------------------------+
+        | 'actual_production'  | System energy production. Same units as      |
+        |                      | 'lost_production'.                           |
+        +----------------------+----------------------------------------------+
+        | 'availability'       | Energy-weighted system availability as a     |
```

Comment:
This is just an FYI - NERC is developing reporting requirements for solar generators > 20MW capacity. In that context, "availability" and "outage" have very specific meanings. Down the road there will be a need for tools that compute the NERC-required metrics.

Reply:
Let me see if I can find and share the draft spec